src/components/ec-datepicker/ec-datepicker.vue

```diff
@@ -0,0 +1,205 @@
+    :is-warning="isWarning"
+    :error-message="errorMessage"
+    :disabled="isDisabled"
+    v-on="$listeners"
+    @input-icon-clicked="openCalendar()"
+  />
+</template>
+
+<script>
+import flatpickr from 'flatpickr';
+import { getUid } from '../../utils/uid';
+
+import EcInputField from '../ec-input-field';
+
+export default {
+  name: 'EcDatepicker',
+  components: {
+    EcInputField,
+  },
+  model: {
+    prop: 'value',
+    event: 'value-change',
+  },
+  props: {
+    value: {
+      type: String,
+    },
+    label: {
+      type: String,
+    },
+    note: {
+      type: String,
+    },
+    bottomNote: {
+      type: String,
+    },
+    isWarning: {
+      type: Boolean,
+      default: false,
+    },
+    placeholder: {
+      type: String,
+    },
+    errorMessage: {
+      type: String,
+    },
+    isDisabled: {
+      type: Boolean,
+      default: false,
+    },
+    disabledDates: {
+      type: Object,
+      default: () => {},
+    },
+    areWeekendsDisabled: {
+      type: Boolean,
+      default: false,
+    },
+    options: {
+      type: Object,
+      default: () => ({}),
+    },
+  },
+  data() {
+    return {
+      uid: getUid(),
+    };
+  },
+  computed: {
+    datepickerModel: {
+      get() {
+        return this.value;
+      },
+      set(value) {
+        this.flatpickrInstance.setDate(value);
+        this.$emit('value-change', value);
+      },
+    },
+    id() {
+      return `datepicker-${this.uid}`;
+    },
+  },
+  watch: {
+    disabledDates: {
+      immediate: true,
+      handler(newValue) {
+        if (newValue) {
+          this.disabledDatesMap = new Map(Object.entries(newValue));
+        }
+        if (this.flatpickrInstance) {
+          this.flatpickrInstance.redraw();
+        }
+      },
+    },
+    areWeekendsDisabled() {
+      /* istanbul ignore next */
+      if (this.flatpickrInstance) {
+        this.flatpickrInstance.redraw();
+      }
+    },
+    options(newValue) {
+      const newOptions = this.mergeWithDefaultOptions(newValue);
+      for (const [key, value] of Object.entries(newOptions)) {
+        // https://github.com/ankurk91/vue-flatpickr-component/blob/9.0.5/src/component.js#L186
+        if (!key.match(/on[A-Z]/)) {
+          this.flatpickrInstance.set(key, value);
+        }
+      }
+    },
+  },
+  mounted() {
+    this.flatpickrInstance = flatpickr(this.$refs.input.getInputRef(), this.mergeWithDefaultOptions(this.options));
+    /* istanbul ignore next */
+    if (this.flatpickrInstance.calendarContainer) {
+      this.flatpickrInstance.calendarContainer.dataset.test = 'ec-datepicker__calendar';
+      this.flatpickrInstance.calendarContainer.dataset.relDataTest = `${this.$attrs['data-test']} ec-datepicker`.trim();
+    }
+  },
+  beforeDestroy() {
+    /* istanbul ignore next */
+    if (this.flatpickrInstance) {
+      this.flatpickrInstance.destroy();
+    }
+  },
+  methods: {
+    openCalendar() {
+      this.flatpickrInstance.open();
```

Comment:
It still has different behavior depending on which part of the component we click on to open the calendar. When we click on the input field part and the calendar is already open, it simply remains open. However, when we click on the icon part, the calendar will re-run the opening animation each time.
I understand that the problem is not solved by checking whether the calendar is being displayed, since the calendar is closed when we click on the icon part.
Anyway, I can't really think of a simple way to solve it (without having to touch the input field component), so I guess I could accept this behavior if there is no simple solution for this.

Reply:
Hi Jose,  I am aware of this behaviour but I felt this solution is good enough. I wouldn't like to overcomplicate things since it does what we want now, it opens the datepicker when you click on the icon. If you are not strong opiniated on this I would like to keep it as is.